src/main/java/com/parkhomovsky/bookstore/service/implementation/OrderServiceImpl.java

```diff
@@ -0,0 +1,153 @@
+package com.parkhomovsky.bookstore.service.implementation;
+
+import com.parkhomovsky.bookstore.dto.order.OrderDto;
+import com.parkhomovsky.bookstore.dto.order.OrderPlaceRequestDto;
+import com.parkhomovsky.bookstore.dto.order.OrderUpdateStatusRequest;
+import com.parkhomovsky.bookstore.dto.order.StatusUpdateResponseDto;
+import com.parkhomovsky.bookstore.dto.orderitem.OrderItemDto;
+import com.parkhomovsky.bookstore.enums.Status;
+import com.parkhomovsky.bookstore.exception.EmptyShoppingCartException;
+import com.parkhomovsky.bookstore.exception.EntityNotFoundException;
+import com.parkhomovsky.bookstore.mapper.OrderItemsMapper;
+import com.parkhomovsky.bookstore.mapper.OrderMapper;
+import com.parkhomovsky.bookstore.model.CartItem;
+import com.parkhomovsky.bookstore.model.Order;
+import com.parkhomovsky.bookstore.model.OrderItem;
+import com.parkhomovsky.bookstore.model.ShoppingCart;
+import com.parkhomovsky.bookstore.model.User;
+import com.parkhomovsky.bookstore.repository.order.OrderRepository;
+import com.parkhomovsky.bookstore.repository.orderitems.OrderItemsRepository;
+import com.parkhomovsky.bookstore.service.OrderService;
+import com.parkhomovsky.bookstore.service.ShoppingCartService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final ShoppingCartService shoppingCartService;
+    private final OrderRepository orderRepository;
+    private final OrderItemsRepository orderItemsRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemsMapper orderItemsMapper;
+
+    public OrderDto process(OrderPlaceRequestDto orderPlaceRequestDto,
```

Comment:
```suggestion
    @Transactional
    @Override
    public OrderDto process(OrderPlaceRequestDto orderPlaceRequestDto,
```
And think in which other methods you need to add `Transactional`